data/functions.sql

```diff
@@ -105,8 +108,13 @@ BEGIN
                                     'hostel', 'hotel', 'motel', 'museum')
                    OR railway_val IN ('subway_entrance')) THEN 17
              WHEN (amenity_val IN ('bench', 'waste_basket')) THEN 18
-             ELSE NULL END
-    );
+             ELSE NULL END;
+  RETURN (CASE
+    WHEN way_area > 1.0e7 THEN zoom - 4
+    WHEN way_area > 1.0e6 THEN zoom - 3
+    WHEN way_area > 1.0e5 THEN zoom - 2
+    WHEN way_area > 1.0e4 THEN zoom - 1
+    ELSE zoom END);
```

Comment:
I don't know - given we switch from NE to OSM at 9 (soon to be 8) do we want to show some of this stuff at pre-OSM zooms? I reckon that might be a decision to punt on until we get some feedback from @nvkelso on the new way POIs are coming in as zoom changes.